### PR TITLE
fix: hide PromQL from table panel type

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Removed PromQL tab from table panel type

#### Related Issues / PR's

https://github.com/SigNoz/engineering-pod/issues/1441

#### Screenshots

https://github.com/user-attachments/assets/57c60b28-a084-43de-b167-1ed7e6f28094

#### Affected Areas and Manually Tested Areas

- tested dashboard edit mode with different panel types
- tested panel type switch and default query-type selection - (like should switch to query-builder type when switching to panel type to table panel)

### CODE REVIEW

**Review by rahulkeswani101 — DISMISSED** [review]

File: frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx [hunk 1]
```diff
@@ -168,51 +170,63 @@ function QuerySection({
...
+						<Terminal size={14} />
+						<Typography>ClickHouse Query</Typography>
+					</Button>
+				),
+				tab: <Typography>ClickHouse Query</Typography>,
+				children: <ClickHouseQueryContainer />,
+			},
+		];
+
+		if (selectedGraph !== PANEL_TYPES.TABLE) {
```
Comment (vikrantgupta25) [inline frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx:205]: NIT: better to maintain a map of supported types vs panel types , easy to read and scalable to maintain

**Review by vikrantgupta25 — DISMISSED** [review]

File: frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx [hunk 2]
```diff
@@ -223,6 +211,16 @@ function QuerySection({
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [handleRunQuery]);
 
+	useEffect(() => {
+		// switch to query builder if query type is not supported
+		if (
+			selectedGraph === PANEL_TYPES.TABLE &&
```
Comment (vikrantgupta25) [inline frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx:217]: qq : this is to handle existing promQL query dashboards ?

**Review by vikrantgupta25 — APPROVED** [review]
LGTM

File: frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/src/container/NewWidget/LeftContainer/QuerySection/index.tsx:217]: yes, like someone is coming from promql (in lets say time-series) to a table panel, then we will smartly switch to query-builder

**Review by rahulkeswani101 — APPROVED** [review]

**Pull request merged**